### ISSUE TITLE
Any chance to add MathJaX support in addition to KaTeX?

### ISSUE THREAD

**Issue by yzhang-gh** [yzhang-gh/vscode-markdown#531]
Any chance to add MathJaX support in addition to KaTeX?
Like letting the user to select the Math Rendering Engine?

MathJaX is almost fully compliant with LaTeX while KaTeX misses few abilities.

_Originally posted by @RoyiAvital in https://github.com/yzhang-gh/vscode-markdown/issues/106#issuecomment-468934680_

**Comment by yzhang-gh** [comment]
Hi @RoyiAvital, thanks for the feedback.

There is a workaround now (use MathJax scripts and export it to HTML). For example,

```markdown
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script id="MathJax-script" async src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>

\\(\alpha\\)
```

Remember to use `\\(` and `\\[` rather than `$` and `$$`.

The only disadvantage is you cannot see the preview inside VSCode as it disables scripts for security concerns.
I suggest using `markdown.extension.print.onFileSave` option and [Live Server](https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer) extension.

![mathjax](https://user-images.githubusercontent.com/7588612/66620519-61bfbc00-ec13-11e9-81c5-985e608228cd.gif)

**Comment by RoyiAvital** [comment]
As I wrote in the other issue, I really think this feature will make this extension perfect for students as alternative to LaTeX.
Yet in that case KaTeX won't cut it. it has too many limitations and missing features.

It will be great to have MathJaX support in addition to KaTeX.
As the other features in this package are amazing (I like specifically all the flexibility with TOC).

**Comment by PierreMarchand20** [comment]
Just out of curiosity, what are the missing features of KaTeX ? 

You could also ask them to add it, I mean the advantage of KaTeX is that it is really fast, which may be convenient while taking notes.

**Comment by RoyiAvital** [comment]
On modern Chrome the speed difference isn't an issue.
MathJaX has support for all environments of LaTeX while KaTeX doesn't and misses a lot of them.
It also doesn't hgave the reference and labeling system of MathJaX.

We have written many times about features of KaTeX but they are not easy to implement in KaTeX. I guess due to the design choices made there.

If you want to have a true replacement for LaTeX in MarkDown, you need MathJaX.

**Comment by yzhang-gh** [comment]
@RoyiAvital I agree it might be a good addition. It is just because I don't much time for it. BTW, we haven't got a new version of this extension for near 3 months 😅.

**Comment by RoyiAvital** [comment]
@yzhang-gh , I see new features in the extension (Great!).
Any chance to give this another thought?

It is the only blocker for me to use it as my daily MarkDown driver for University.

Thank You.

**Comment by yzhang-gh** [comment]
Actually, most of those features come from PRs (or only require a few tweaks to the source code).

MathJax support is not a "small" feature. It requires extra dependency and needs testing.
For comparison, supporting `\(\)` and `\[\]` delimiters seems to be a more popular feature (#552) and easier to implement. (However, no ETA too. 😑)

I would suggest using [Markdown+Math](https://marketplace.visualstudio.com/items?itemName=goessner.mdmath) plugin which is more dedicated to math support.
I'm not sure whether it is compatible with this extension. But I can add an option to disable math support of this one. It is much easier.

**Comment by RoyiAvital** [comment]
Markdown+Math use KaTeX as well instead of MathJaX.

Can't the MathJaX functionality be borrowed from other Extensions?
[MarkDown Preview Enhanced](https://github.com/shd101wyy/markdown-preview-enhanced) is using MathJaX. Could their code be borrowed?

**Comment by yzhang-gh** [comment]
> Markdown Preview Enhanced is using MathJaX. Could their code be borrowed?

It goes for an entirely different direction, providing a standalone preview, rather than contribute to the VSCode's default one.

Since you have mentioned it, why not just use it 😳?
It is compatible with this extension (it has its own preview window and supports exporting to PDF/HTML)

**Comment by RoyiAvital** [comment]
I am using it. But your extension has many features I'd like to have.

**Comment by yzhang-gh** [comment]
Yep, I have seen many people using the two extensions together. This extension provides the general features and the MPE enables many useful Markdown dialects.

**Comment by RoyiAvital** [comment]
I'm not sure I'm following.
Which one will render the output, this one or MPE?
MPE uses MathJaX (Can be configured) while this uses the limited KaTeX.
When both implement the same feature, which one will be the effective one?

**Comment by yzhang-gh** [comment]
https://shd101wyy.github.io/markdown-preview-enhanced/#/usages
![image](https://user-images.githubusercontent.com/7588612/75134323-e28dd900-5718-11ea-9911-e6102ffd1031.png)

For this extension, it is `ctrl+shift+v`/`ctrl+k v`.

**Comment by RoyiAvital** [comment]
I can guarantee you that `Ctrl + k, v` is what's used to enable the preview for MPE on Visual Studio Code:

![image](https://user-images.githubusercontent.com/7347975/75135631-d2f79b80-56ea-11ea-8eda-595bbaa7d2b6.png)

**Comment by yzhang-gh** [comment]
Alright, you may not be very familiar with VSCode (and the MPE's docs are not clear) so I tried it out for you.

You need to invoke the toggle-preview command from the command palette because the assigned keybindings are conflicted.

![image](https://user-images.githubusercontent.com/7588612/75136309-3e5b6080-571f-11ea-9e84-144ccd8a3577.png)

![image](https://user-images.githubusercontent.com/7588612/75136086-7ada8c80-571e-11ea-9429-594285fff5a5.png)

You can reassign those keybindings as you wish
(see https://code.visualstudio.com/docs/getstarted/keybindings).

**Comment by PeterWang-dev** [comment]
Any update about this function?

**Comment by mathematicsformisfits** [comment]
Adding to this request.  MathJax has full accessibility support, KaTeX has none.
